lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1942,6 +1942,13 @@ sub handle_retry ($self) {
     return 1;
 }
 
+sub enqueue_restart ($self) {
+    return undef unless my $gru = eval { OpenQA::App->singleton->gru };    # gru might not be present within tests
```

Comment:
This is rather ugly, does it affect that many tests that its better than a mock?

Reply:
I've actually just copied it from another function. Let's see what breaks without it.